pkg/operation/botanist/health_check.go

```diff
@@ -488,12 +489,17 @@ func (b *HealthChecker) CheckMonitoringControlPlane(
 		return exitCondition, nil
 	}
 
-	statefulSetList, err := statefulSetLister.StatefulSets(namespace).List(monitoringSelector)
+	requiredMonitoringStatefulSets := sets.NewString(common.PrometheusStatefulSetName)
```

Comment:
Please move this out to a separate method like we did it for `computeRequiredControlPlaneDeployments` and put the stateful sets that always apply into a `var` 'constant'.